src/context/note-context.js

```diff
@@ -64,7 +76,7 @@ const NoteContextProvider = ({children}) => {
     
 
 
-    return (<NoteContext.Provider value={{state , dispatch ,notes , isLogin , setIsLogin , addNoteToBackend }}>{children}</NoteContext.Provider>)
+    return (<NoteContext.Provider value={{deleteNote, state , pinNotes, setPinNotes, dispatch, notes, setNotes, isLogin, setIsLogin, addNoteToBackend }}>{children}</NoteContext.Provider>)
 }
```

Comment:
Suggestion: I don't think you need to pass api calls using context. You can call them directly.